Decred Wallet/Features/TransactionDetails/TransactionDetailsViewController.swift

```diff
@@ -8,262 +8,308 @@ import UIKit
+        if section == 1 {
+            return self.generalTxDetails.count
+        } else if section == 2 {
+            return self.isTxInputsCollapsed ? 0 : transaction.inputs.count
+        } else if section == 3 {
+            return self.isTxOutputsCollapsed ? 0 : transaction.outputs.count
+        } else {
+            return 1
+        }
     }
-    
+
     func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
-        return 0.5
+        if section == 2 || section == 3 {
+            return 48
+        } else {
+            return 1
+        }
     }
-    
+
     func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
-        let headerView = UIView.init(frame: CGRect.zero)
-        headerView.backgroundColor = #colorLiteral(red: 0.8039215803, green: 0.8039215803, blue: 0.8039215803, alpha: 1)
-        headerView.frame.size.height = 0
+        if section == 0 {
+            return UIView.init(frame: CGRect.zero)
+        } else if section == 2 {
+            return self.inputOutputSectionHeaderView(section: section,
+                                                        title: String(format: LocalizedStrings.inputsConsumed, transaction.inputs.count),
+                                                        isCollapsed: self.isTxInputsCollapsed)
+        } else if section == 3 {
+            return self.inputOutputSectionHeaderView(section: section,
+                                                        title: String(format: LocalizedStrings.outputsCreated, transaction.outputs.count),
+                                                        isCollapsed: self.isTxOutputsCollapsed)
+        } else {
+            let headerView = UIView.init(frame: CGRect.zero)
+            headerView.backgroundColor = UIColor.appColors.gray
+            headerView.frame.size.height = 1
+            return headerView
+        }
+    }
+
+    private func inputOutputSectionHeaderView(section: Int,
+                                                 title: String,
+                                                 isCollapsed: Bool) -> UIView {
+        let headerView = UIView.init(frame: CGRect(x: 0, y: 0, width: self.transactionDetailsTable.frame.size.width, height: 48))
+        headerView.backgroundColor = UIColor.white
+        headerView.frame.size.height = 48
+        headerView.horizontalBorder(borderColor: UIColor.appColors.gray, yPosition: 0, borderHeight: 1)
+
+        let headerLabel = UILabel.init(frame: CGRect(x: 16, y: 1, width: self.view.frame.size.width - 56, height: 47))
+        headerLabel.textColor = UIColor.appColors.bluishGray
+        headerLabel.font = UIFont(name: "SourceSansPro-Regular", size: 14)
+        headerLabel.numberOfLines = 1
+        headerLabel.text = title
+        headerView.addSubview(headerLabel)
+
+        let arrowImageView = UIImageView.init(frame: CGRect(x: self.view.frame.size.width - 40, y: 12, width: 24, height: 24))
```

Comment:
Should probably use `self.transactionDetailsTable.frame.size.width` instead of `self.view.frame.size.width`. Seeing as this max width value is used in multiple places in this function, it may be a good idea to declare a `maxHeaderWidth` let constant `= self.transactionDetailsTable.frame.size.width`.